enumerables.rb

```diff
@@ -0,0 +1,153 @@
+module Enumerable
+  def my_each
+    return to_enum :name unless block_given?
+
+    for element in self
+      yield element
+    end
+  end
+
+  def my_each_with_index
+    return to_enum :name unless block_given?
+
+    c = 0
+    for element in self
+      yield element, c
+      c += 1
+    end
+  end
+
+  def my_any?(args = nil)
+    if args.nil?
+      if block_given?
+        my_each { |x| return true if yield(x) }
+      else
+        my_each { |x| return true if x }
+      end
+    elsif args.is_a?(Regexp)
+      my_each { |x| return true if x.match(args) }
+    elsif args.is_a?(Module)
+      my_each { |x| return true if x.is_a?(args) }
+    else
+      my_each { |x| return true if x == args }
+    end
+    false
+  end
+
+  def my_none?(args = nil)
+    if args.nil?
+      if block_given?
+        my_each { |x| return false if yield(x) }
+      else
+        my_each { |x| return false if x }
+      end
+    elsif args.is_a?(Regexp)
+      my_each { |x| return false if x.match(args) }
+    elsif args.is_a?(Module)
+      my_each { |x| return false if x.is_a?(args) }
+    else
+      my_each { return false if val == args }
+    end
+    true
+  end
+
+  def my_select
+    return to_emum :my_select unless block_given?
+
+    new_hash = {}
+    new_array = []
+
+    if is_a?(Hash)
+      my_each do |x, y|
+        new_hash[x] = y if yield(x, y)
+      end
+      return new_hash
+    else
+      my_each do |x|
+        new_array.push(x) if yield(x)
+      end
+    end
+    new_array
+  end
+
+  def my_all?(args = nil)
+    return to_enum unless block_given? || !args.nil?
+
```

Comment:
At the my_all?, we got these errors:

1. - rspec ./spec/enumerables_spec.rb:161 # enumerables #my_all? when no block or argument is given returns true when none of the collection members are false or nil
2. - rspec ./spec/enumerables_spec.rb:165 # enumerables #my_all? when no block or argument is given returns false when one of the collection members are false or nil
3. - rspec ./spec/enumerables_spec.rb:202 # enumerables #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern